### DISCUSSION TITLE
Adding permissions across remote databases

### DISCUSSION THREAD

**Discussion by willtpwise** [hasura/graphql-engine#10075] [Question]
I'm exploring the Hasura remote database feature and looking to see if it is possible to write a permission statement across two databases. 

## Table structure

**First Database**
```
users
  id: uuid
  account_id: uuid

accounts
  id: uuid
```

**Second Database**
```
bookings
  id: uuid
  account_id: uuid
```

## Relationships
`bookings.account_id` -> _object relationship_ -> `accounts.id`
`users.account_id` -> _object relationship_ -> `accounts.id`

## Bookings permission model
I'd like to define permissions across these two tables and confirm that the current user belongs to the account for this booking. 

For example:

```
{ account: { users: { id: { _eq: "[X-Hasura-User-Id]" } } }
```

Unfortunately, "account" is not available as an option in the permissions relationships dropdown. 

## Notes
* Whilst this undoubtedly will come with some hefty performance draw backs I'm comfortable with those in theory
* I've created the appropriate relationships between the remote tables
* Both databases are postgres
* There's a flag for `HASURA_GRAPHQL_ENABLE_REMOTE_SCHEMA_PERMISSIONS` but I believe that's for remote "schemas" and not remote "databases".

**Comment by willtpwise** [comment]
I've also been reading this article, which suggest this is possible but unfortunately I'm not seeing the related table when creating a permisson:

https://hasura.io/blog/introducing-remote-table-authorization-write-permission-rules-across-all-your-data/

**Reply by manasag** [reply]
It should work as per the blog post instructions. Can you share screenshots of the relationship definition and the permission view where you don't see the relationship in the Row select permission builder.

**Reply by willtpwise** [reply]
Thanks @manasag. Apologies for the delay.

I've noticed that the relationships work when the feature flag for `Data Tab Performance Mode`. 

Here's a quick walk through video explaining the problem. 
https://www.loom.com/share/f4b9a81fe9f04298b477f0264548d64d

For your benefit, I'm on v2.36.1 running via Docker.

**Reply by willtpwise** [reply]
Furthermore, there appears to be an issue accessing nested relationships in foreign database permissions. 

See this walkthrough: https://www.loom.com/share/4a83bcbbbe0443edbe34ae706e83c503

What I'm seeing is that when the `Data Tab Performance Mode` feature is enabled, accessing the foreign relationship in the permission builder is possible, but accessing that relationship's own relationships (i.e. two steps from the source) is not possible. 

Would love it if you have an idea on a workaround. 🙌
